src/test/builder/safe.redux.builder.ts

```diff
@@ -89,9 +89,7 @@ export const aMinedSafe = async (
     form[getOwnerAddressBy(i)] = accounts[i]
   }
 
-  const openSafeProps = await createSafe(form, accounts[0])
-
-  return openSafeProps.safeAddress
+  return createSafe(form, accounts[0]).then((receipt) => receipt.events?.ProxyCreation.returnValues.proxy)
```

Comment:
why are you doing this? `createSafe` already returns the safeAddress. https://github.com/gnosis/safe-react/pull/1437/files#diff-b5f8a9922c89630abeb184e2ec4d2539R75

Reply:
Actually, originally was returning the `safeAddress`.

But then, somewhere in the middle `createSafe` return value was changed to return the `promiEvent` instead (probably because of https://github.com/gnosis/safe-react/blob/29d3b2ca3211d4d0252d61616bdab5206b6356b9/src/routes/open/container/Open.tsx#L134)

In fact, I'm not sure why the whole `then` block (https://github.com/gnosis/safe-react/blob/29d3b2ca3211d4d0252d61616bdab5206b6356b9/src/routes/open/container/Open.tsx#L70-L75) remained there, as it was useless.

What I wanted, was to fix the typing errors that arose after typing the contracts. If we all agree that the `then` block is useless, I can remove it.

Reply:
the promise is returning `return { safeAddress: safeProps.address, safeTx: receipt }` so in the `then` you should be able to get an object with `{safeAddress and safeTx}` isn't it?

Reply:
For that end, it should have been
```typescript
return promiEvent.on(/*...*/).then(/*...*/).catch(/*...*/)
```

instead of
```typescript
promiEvent.on(/*...*/).then(/*...*/).catch(/*...*/)
return promiEvent
```
as it's now.

But if you do the first, you lose the `promiEvent` that is needed in https://github.com/gnosis/safe-react/blob/29d3b2ca3211d4d0252d61616bdab5206b6356b9/src/routes/opening/index.tsx#L171-L177

Reply:
make sense, feel free to remove the code in the `then` in Open.tsx